packages/core/src/components/segmented-control/segmentedControl.tsx

```diff
@@ -83,133 +83,151 @@ export interface SegmentedControlProps
     small?: boolean;
 }
 
+// This allows the ability to pass a more strict type for `options`/`onValueChange`
+// i.e. <SegmentedControl<Intent> />
+interface ReactFCWithGeneric extends React.FC<SegmentedControlProps> {
```

Comment:
Nit this is a specific interface for `SegmentedControl` so lets name it something in accordance with that.
```suggestion
interface GenericSegmentedControl extends React.FC<SegmentedControlProps> {
```

Reply:
✔️

Reply:
Also just as an FYI, I created this type to match the fact that all other components in this repo specify the type `React.FC`. But really you don't need to manually add a type at all, you could just have 

```tsx
export const SegmentedControl = React.forwardRef(
    <T extends string>(props: SegmentedControlProps<T>, ref: React.ForwardedRef<HTMLDivElement>) => {
```

Reply:
Renamed and commented to match the other cases.